brainglobe_atlasapi/atlas_generation/atlas_scripts/demba_dev_mouse.py

```diff
@@ -0,0 +1,345 @@
+    """
+    base_path = f"{download_dir_path}/{METADATA.name}/"
+    if modality == "stpt":
+        reference_path = (
+            f"{base_path}DeMBA_templates/DeMBA_P{age}_brain.nii.gz"
+        )
+        annotation_path = (
+            f"{base_path}AllenCCFv3_segmentations/20um/2022/"
+            f"DeMBA_P{age}_segmentation_2022_20um.nii.gz"
+        )
+        volume_resolution = 20
+    elif modality == "allen_stpt":
+        reference_path = (
+            f"{base_path}allen_stpt_10um/DeMBA_P{age}_AllenSTPT_10um.nii.gz"
+        )
+        annotation_path = (
+            f"{base_path}AllenCCFv3_segmentations/10um/2022/"
+            f"DeMBA_P{age}_segmentation_2022_10um.nii.gz"
+        )
+        volume_resolution = 10
+
+    elif modality == "mri":
+        reference_path = f"{base_path}mri_volumes/DeMBA_P{age}_mri.nii.gz"
+        annotation_path = (
+            f"{base_path}AllenCCFv3_segmentations/20um/2022/"
+            f"DeMBA_P{age}_segmentation_2022_20um.nii.gz"
+        )
+        volume_resolution = 25
+
+    elif modality == "lsfm":
+        reference_path = f"{base_path}lsfm_volumes/DeMBA_P{age}_lsfm.nii.gz"
+        annotation_path = (
+            f"{base_path}AllenCCFv3_segmentations/20um/2022/"
+            f"DeMBA_P{age}_segmentation_2022_20um.nii.gz"
+        )
+        volume_resolution = 25
+    else:
+        raise ValueError(f"Unknown modality: {modality}")
+    return reference_path, annotation_path, volume_resolution
+
+
+def retrieve_reference_and_annotation(
+    download_dir_path, age, resolution, modality
+):
+    """
+    Retrieve the desired reference and annotation as two numpy arrays.
+
+    Returns:
+        tuple: A tuple containing two numpy arrays. The first array is the
+        reference volume, and the second array is the annotation volume.
+    """
+    reference_path, annotation_path, volume_resolution = (
+        get_reference_and_annotation_paths(download_dir_path, age, modality)
+    )
+    annotation = load_any(annotation_path)
+    reference = load_any(reference_path)
+    zoom_factors = tuple(volume_resolution / resolution for _ in range(3))
+    reference = zoom(reference, zoom_factors, order=1)
+    if annotation.shape != reference.shape:
+        """
+        we unfortunately did not provide 25um segmentations so
+        we will just downsample the 20um ones
+        """
+        zoom_factors = tuple(
+            ref_dim / ann_dim
+            for ref_dim, ann_dim in zip(reference.shape, annotation.shape)
+        )
+        annotation = zoom(annotation, zoom_factors, order=0)
+    return reference, annotation
+
+
+def retrieve_additional_references(
+    download_dir_path, age, resolution, modalities
+):
+    """This function only needs editing if the atlas has additional reference
+    images. It should return a dictionary that maps the name of each
+    additional reference image to an image stack containing its data.
+    """
```

Comment:
maybe a quick explanation of the relationship between the main and additional references here, instead of the boilerplate comment?